testutil/network/basenet/accounts.go

```diff
@@ -0,0 +1,194 @@
+package basenet
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	testcli "github.com/cosmos/cosmos-sdk/testutil/cli"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/network"
+	"github.com/pokt-network/poktroll/testutil/testkeyring"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+const warnNoModuleGenesisFmt = "WARN: no %s module genesis state found, if this is unexpected, ensure that genesis is populated before creating on-chain accounts"
+
+// CreateKeyringAccounts populates the Keyring associated with the in-memory
+// network with memnet.numKeyringAccounts() number of pre-generated accounts.
+func (memnet *BaseInMemoryNetwork) CreateKeyringAccounts(t *testing.T) {
+	t.Helper()
+
+	// Keyring MAY be provided setting InMemoryNetworkConfig#Keyring.
+	if memnet.Config.Keyring == nil {
+		t.Log("Keyring not initialized, using new in-memory keyring")
+
+		// Construct an in-memory keyring so that it can be populated and used prior
+		// to network start.
+		memnet.Config.Keyring = keyring.NewInMemory(memnet.Config.CosmosCfg.Codec)
+	} else {
+		t.Log("Keyring already initialized, using existing keyring")
+	}
+
+	// Create memnet.NumKeyringAccounts() number of accounts in the configured keyring.
+	accts := testkeyring.CreatePreGeneratedKeyringAccounts(
+		t, memnet.Config.Keyring, memnet.Config.GetNumKeyringAccounts(t),
+	)
+
+	// Assign the memnet's pre-generated accounts to a new pre-generated
+	// accounts iterator containing only the accounts which were also created
+	// in the keyring.
+	memnet.PreGeneratedAccountIterator = testkeyring.NewPreGeneratedAccountIterator(accts...)
+}
+
+// CreateOnChainAccounts creates on-chain accounts (i.e. auth module) for the sum of
+// the configured number of suppliers, applications, and gateways.
```

Comment:
For my own understanding this ensure their public keys are available even without being interacted with? If not #PUC

Reply:
Good call out, I've renamed this to `FundOnChainAccounts` to improve clarity.